14/umbraco-cms/reference/umbraco-flavored-markdown.md

```diff
@@ -0,0 +1,103 @@
+
+{% hint style="info" %}
+**What is Markdown?**
+If you are unfamiliar with Markdown, you can read more about its philosophy and syntax on the Daring Fireball website.
+<https://daringfireball.net/projects/markdown/syntax>
+{% endhint %}
+
+By using Markdown for labels means that as well as offering basic text formatting, it natively supports the use of HTML. This means that web components can be used to handle more complex label templating scenarios.
+
+UFM is built on top of [GitHub Flavored Markdown](https://github.github.com/gfm/) and [CommonMark](https://spec.commonmark.org/) specifications. The implementation for Umbraco 14 has been developed as an extension to the [Marked library](https://marked.js.org/).
+
+
+## Syntax
+
+The essence of the UFM syntax is curly brackets with a marker prefix.
+
+```
+{<marker prefix> <contents> }
+```
+
+For clarity...
+
+- The opening is `{` U+007B Left Curly Bracket
+- The closing is `}` U+007D Right Curly Bracket
+- The marker prefix is any valid Unicode character(s)
+- The remaining contents inside the curly brackets can contain any Unicode, including whitespace
+
+An example of this syntax to render a value of a property by its alias would be: `{= bodyText }`.
+
+The curly brackets indicate that the UFM syntax should be processed. The `=` marker prefix indicates which UFM component should be rendered, and the `bodyText` contents is the parameter that is passed to that UFM component.
+
+With this example, the syntax `{= bodyText }` would be processed and rendered as the following markup:
+
+```
+<ufm-label-value alias="bodyText"></ufm-label-value>
+```
+
+The internal working of the `ufm-label-value` component would then be able to access the property's value using the [Context API](../extending/backoffice-setup/working-with-data/context-api).
+
+
+## Available UFM components
+
+As for Umbraco 14.1.0, the following UFM components are available to use.
+
+| Name        | Marker | Example syntax    | Renders component                    |
+| ----------- | ------ | ----------------- | ------------------------------------ |
+| Label Value | `=`    | `{=bodyText}`     | `<ufm-label-value alias="bodyText">` |
+| Localize    | `#`    | `{#general_name}` | `<umb-localize key="general_name">`  |
+
+
+More UFM components will be available in upcoming Umbraco releases.
+
+If you wish to develop your own custom UFM component, you can use the `ufmComponent` extension type:
+
+```
+{
+	type: 'ufmComponent',
+	alias: 'My.Markdown.CustomComponent',
+	name: 'My Custom Markdown Component',
+	api: () => import('./components/my-custom.component.js'),
+	meta: {
+		marker: '%',
+	},
+}
+```
+
+The corresponding JavaScript/TypeScript API would contain a method to render the custom label/markup.
+
+```
+export class MyCustomComponent implements UfmComponentBase {
```

Comment:
This would need to extend UmbLitElement, right?

```suggestion
export class MyCustomComponent extends UmbLitElement implements UfmComponentBase {
```

Reply:
Oh, I see. It's a UFM component. Should we perhaps name the example here in the docs something else?

Reply:
I'm thinking `MyCustomUfmComponentApi`.

I'd also noticed a typo with the `UfmComponentBase` base class, (it needed the "Umb" prefix). 🤦 
But also, I'm wondering whether `UmbUfmComponentBase` is a suitable name? As it implements the `UmbUfmComponentApi` interface, so maybe the base class should be renamed to `UmbUfmComponentApiBase`? 🤔

Reply:
I thought we agreed to allow the "Ufm" namespace 😅